### PR TITLE
Add libde265 h.265 video codec

### PR DESCRIPTION
**Pull request by frmdstryr**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by frmdstryr** [comment]
@conda-forge/help-c-cpp, ready for review!

File: recipes/libde265/run_test.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+git clone https://github.com/strukturag/libde265-data.git
+dec265 -q -c -f 100 ./libde265-data/IDR-only/paris-352x288-intra.bin
+dec265 -t 4 -q -c -f 100 ./libde265-data/IDR-only/paris-352x288-intra.bin
+dec265 -0 -q -c -f 100 ./libde265-data/RandomAccess/paris-ra-wpp.bin
+dec265 -0 -t 4 -q -c -f 100 ./libde265-data/RandomAccess/paris-ra-wpp.bin
```
Comment (carterbox) [inline recipes/libde265/run_test.sh:5]: These tests are mainly to ensure that the libraries were packages correctly. Do your binaries have a `--help` or `--version` option that will just print and return 0? If so, please do that for all the binaries: encode, decode, and hdrcopy. Full unit tests are not really necessary.

File: recipes/libde265/run_test.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+git clone https://github.com/strukturag/libde265-data.git
```
Comment (carterbox) [inline recipes/libde265/run_test.sh:1]: In addtion to testing that your binaries can load. Please use [existence tests](https://conda-forge.org/docs/maintainer/adding_pkgs.html#simple-existence-tests) to explicitly check that the following were installed to the correct place:

- all libs (both .lib and .dll on Windows)
- one of the cmake files
- one of the headers

These check will prevent accidental removals or ABI breaking name changes from going un-noticed in the future.

File: recipes/libde265/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/strukturag/libde265/releases/download/v{{ version }}/libde265-{{ version }}.tar.gz
+  sha256: 29bc6b64bf658d81a4446a3f98e0e4636fd4fd3d971b072d440cef987d5439de
+
+build:
+  number: 0
+
```
Comment (carterbox) [inline recipes/libde265/meta.yaml:14]:
```suggestion
build:
  number: 0
  run_exports:
    - {{ pin_subpackage('libde256', max_pin='x.x.x') }}

```
Run exports ensure that the correct version is exported to the run dependencies of packages built against this package. Please adjust the *API* max_pin for the *ABI* compatability of this library. If you don't know what at what versions the ABI will break, pinning to 'x.x.x' is sufficiently safe.

**Review by carterbox — CHANGES_REQUESTED** [review]
Except for below, recipe looks good.

File: recipes/libde265/meta.yaml (hunk 3 above)
Comment (frmdstryr) [reply recipes/libde265/meta.yaml]: Didn't know about run_exports, it is pinned as suggested now.

**Comment by frmdstryr** [comment]
Thank you @carterbox !

**Pull request merged**